File: TaskCpuArrivalEvent.cpp
//
// Created by Can Candan on 04/06/17.
//

#include "TaskCpuArrivalEvent.h"
#include "Cpu.h"
#include "CpuInEvent.h"

TaskCpuArrivalEvent::TaskCpuArrivalEvent(Simulation *pSimulation, Task *pTask, double time) : Event(pSimulation, pTask,
                                                                                                 time) {
    name="TaskCpuArrivalEvent";

}

void TaskCpuArrivalEvent::process() {
    log();
    Cpu* cpu= sim->getFirstIdleCpu();
    if (cpu) {
        Event* event= new CpuInEvent(sim, task, time, cpu);
        sim->schedule(event);
    } else {
        sim->sjfQueue.push(task);
    }
}

File: TaskCpuArrivalEvent.java
/**
 * Created by veraxmedax on 03/06/2017.
 */
public class TaskCpuArrivalEvent extends Event {

    public TaskCpuArrivalEvent(Simulation sim, Task task, double time) {
        super(sim, task, time);
    }

    @Override
    void process() {
        Cpu cpu = getSim().getFirstIdleCpu();
        if (cpu != null) {
            Event cpuInEvent = new CpuInEvent(getSim(), getTask(), getTime(), cpu);
            getSim().schedule(cpuInEvent);
        } else {
            getSim().getSjfQueue().add(getTask());
        }
    }
}
